lib/rucio/daemons/replicarecoverer/suspicious_replica_recoverer.py

```diff
@@ -80,7 +82,54 @@ def check_suspicious_policy(policy: dict[str, str], file_metadata_datatype: str,
     return action
 
 
-def declare_suspicious_replicas_bad(once: bool = False, younger_than: int = 5, nattempts: int = 5, vos: Optional[list[str]] = None, limit_suspicious_files_on_rse: int = 5, json_file_name: str = "/opt/rucio/etc/suspicious_replica_recoverer.json", sleep_time: int = 3600, active_mode: bool = False) -> None:
+def parse_replica_datatype(
+        did_name_expression: Optional[str],
+        file_name: str,
+        rse_key: str,
+        logger: Callable) -> Optional[str]:
+
+    file_metadata_datatype = None
+    if did_name_expression is not None:
+        logger(logging.WARNING, f'Using {did_name_expression} as regex to deterime file datatype for file: {file_name} on rse: {rse_key}')
+        try:
+            file_metadata_datatype_reg = re.compile(did_name_expression).search(file_name)
+            if file_metadata_datatype_reg is not None:
+                file_metadata_datatype = file_metadata_datatype_reg[0]
+        except (re.error, TypeError) as e:
+            logger(logging.WARNING, f'Could not deterimine datatype using regex {did_name_expression} on file: {file_name}, ignoring: {e}')
+
+    return file_metadata_datatype
+
+
+def _read_from_config(logger: Callable) -> tuple[str, bool, Optional[str]]:
```

Comment:
```suggestion
def _read_from_config(logger: LoggerFunction) -> tuple[str, bool, Optional[str]]:
```

With LoggerFunction being imported from `rucio.common.types`